Homework4/Alex_Zubok_home_work4_3_crypting_main.cpp

```diff
@@ -0,0 +1,70 @@
+#include "crypting_string.hpp"
+
+int main() {
+  std::cout
+      << "Please enter your string for crypt to 1000 symbols without 'enter': ";
+  char *tmp_string = new char;
+  std::cin.get(tmp_string, 1000, '\n');
+  const int kSize_string =
+      pow(((sqrt(std::strlen(tmp_string)) -
+            static_cast<int>(sqrt(std::strlen(tmp_string)))) > 0)
+              ? static_cast<int>(sqrt(std::strlen(tmp_string))) + 1
+              : static_cast<int>(sqrt(std::strlen(tmp_string))),
+          2);
+  char *string = new char[kSize_string]{};
+  std::strncpy(string, tmp_string, std::strlen(tmp_string));
+  std::cout << "Your string before encrypt is :" << string << std::endl;
+  delete tmp_string;
+  char *tmp = new char[kSize_string - std::strlen(string)];
+  for (size_t i = 0; i < kSize_string - std::strlen(string); i++)
+    *(tmp + i) = '*';
+  std::strncat(string, tmp, kSize_string - std::strlen(string));
+  encrypt_string(string);
+  std::cout << "Your encrypted string is: " << string << std::endl;
+  delete[] string;
+  std::cout << "Please enter your key for decrypt string: ";
+  size_t key{};
+  std::cin >> key;
+  std::cout << "Please enter method of reading string: 1) from file; 2) from "
+               "output: ";
+  size_t method{};
+  std::cin >> method;
+  if (method == 1) {
+    std::cout << "Please enter your file path and filename: ";
+    char *file_str = new char;
+    std::cin.clear();
+    std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+    std::cin.get(file_str, 1000, '\n');
+    std::ifstream File;
+    File.open(file_str, std::ios_base::in);
```

Comment:
```suggestion
    std::ifstream File (file_str); 
```
по дефолту делает open в конструкторе

Reply:
по умолчанию делает для чтения? А для записи указывать нужно?

Reply:
смотрите есть три класса `std::ifstream` открывает по умолчанию для чтения. `std::ifstream` для записи. `std::fstream` для можно совмещать для чтения и записи. Эти классы все используют идиому RAII. Т.е. в конструкторе например открывают файл, а в деструкторе закрывают. Это высокоуровневый принцип RAII

Reply:
Понял, и принял. Код подправил. И почитаю про RAll